utils/utils.go

```diff
@@ -13,3 +13,10 @@ func CloseFile(file *os.File) {
 		log.Panicf("error during closing file %s. %s", err.Path, err.Error())
 	}
 }
+
+// Helper function to log errors
+func FailOnError(err error, msg string) {
+	if err != nil {
+		log.Fatalf("%s: %s", msg, err)
```

Comment:
We dont want to fail on error. NEVER! This is the main assumption of the project. We want always to run, and catch all errors, and in case of one recover to previous safe state